debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java

```diff
@@ -907,4 +920,54 @@ protected Object convertString(Column column, Field fieldDefn, Object data) {
         }
         return super.convertString(column, fieldDefn, data);
     }
+
+    /**
+     *
+     * @param column
+     * @return
+     */
+    @Override
+    protected SchemaBuilder distinctSchema(Column column) {
+        return schemaBuilder(getColumnWithDomainJdbcType(column, typeRegistry.get(column.nativeType())));
+    }
+
+    /**
+     * Provides a ValueConverter that properly resolves the domain type to base type for data of a given column
+     *
+     * @param column the column definition; never null
+     * @param fieldDefn the field definition; never null
+     * @return the value converter to convert the supplied data
+     */
+    @Override
+    protected ValueConverter convertDistinct(Column column, Field fieldDefn) {
+        return converter(getColumnWithDomainJdbcType(column, typeRegistry.get(column.nativeType())), fieldDefn);
+    }
+
+    /**
+     * For a given column and type, traverse type hierarchy and return a column based on base type's JDBC type
+     *
+     * @param column the column
+     * @param postgresType the column's postgres type
+     * @return A new {@link Column} instance with appropriate native JDBC type
+     */
+    private static Column getColumnWithDomainJdbcType(Column column, PostgresType postgresType) {
+        PostgresType baseType = postgresType;
+        while (!baseType.isBaseType()) {
+            baseType = baseType.getBaseType();
+        }
+
+        // This is necessary for situations where PostgresValueConverter delegates schema and converter resolution
+        // to JdbcValueConverters where the resolution is based on the column's jdbcType. For columns that use
+        // domain alias types, this is the OID to the alias type, not the actual base type.
+        //
+        // For example:
+        // CREATE DOMAIN bool2 bool default false;
+        // CREATE TABLE (pk serial, data bool2 not null);
+        //
+        // This guarantees that when the data column's schema and value converter is resolved, it's based on the
+        // fact the resolved base type is bool, not some oid that resolves to an unhandled type.
+        //
+        // Perhaps there are better ways - TBD.
+        return column.edit().jdbcType(baseType.getJdbcId()).nativeType(baseType.getOid()).create();
```

Comment:
Hum, this side-effect of the `get...` method is unexpected. Could this be moved to the construction time?

Reply:
Quite possibly, I'll look and let you know my findings.

Reply:
Another option I explored as well was to instead make a small change to `JdbcValueConverters` where we move what the switch statements are based upon to a method argument and then delegate to those as needed, e.g.

```
    @Override
    public SchemaBuilder schemaBuilder(Column column) {
        return resolveSchemaBuilderByJdbcType(column.jdbcType(), column);
    }

    protected SchemaBuilder resolveSchemaBuilderByJdbcType(int jdbcType, Column column) {
        switch (jdbcType) {
            ...
        }
    }
```
This allows `PostgresValueConverter` to actually call those new resolve methods directly and rather than create a new `Column` instance solely for the sake of controlling the switch argument, we can instead change those methods like so:

```
    @Override
    protected SchemaBuilder distinctSchema(Column column) {
        final int rootTypeJdbcId = typeRegistry.get(column.nativeType()).getRootType().getJdbcId();
        return resolveSchemaBuilderByJdbcType(rootTypeJdbcId, column);
    }
```

I am still going to test your suggestion with Column construction to be sure there are no side in the event you don't particularly care for all this method delegation here.

Reply:
I decided to go with your construction approach as it seemed to solve all the corner nuances without introducing any additional side effects.